client/notifications/action-creators.ts

```diff
@@ -8,23 +12,29 @@ import { AddNotification, MarkNotificationsReadBegin } from './actions'
 export function clearNotifications(): ThunkAction {
   return (dispatch, getState) => {
     const { idToNotification, notificationIds } = getState().notifications
+    const newestNotification = idToNotification.get(notificationIds.last())
+    const timestamp = newestNotification?.createdAt ?? Date.now()
```

Comment:
Passing `Date.now()` isn't safe because this is a local time. The time you've grabbed here also might be a time for a local notification (I think?).

Instead of the code you have here, I'd probably do:
```ts
const newestServerId = notificationIds.findLastEntry(id => {
  const notification = idToNotification.get(id)
  return notification && !notification.local
})
const timestamp = idToNotification.get(newestServerId)?.createdAt
```

(Note that `timestamp` can be `undefined`, but this will be fine with other changes I'll comment)